src/extension/wps/wps-core/src/test/java/org/geoserver/wps/gs/ImportProcessTest.java

```diff
@@ -103,4 +111,42 @@ private void checkBuildings(String result,String expected) throws IOException {
             assertEquals("114", f.getAttribute("FID"));
             assertEquals("215 Main Street", f.getAttribute("ADDRESS"));
 	}
+
+    /**
+     * Test creating a coverage store when a store name is specified but does not exist
+     */
+    @Test
+    public void testCreateCoverageStore() throws Exception {
+        String storeName = SystemTestData.CITE_PREFIX + "raster";
+        // use Coverage2RenderedImageAdapterTest's method, just need any sample raster
+        GridCoverage2D sampleCoverage = Coverage2RenderedImageAdapterTest.createTestCoverage(500, 500, 0,0, 10,10);
+        CoverageStoreInfo storeInfo = catalog.getCoverageStoreByName(storeName);
+        assertNull("Store already exists " + storeInfo, storeInfo);
+        ImportProcess importer = new ImportProcess(getCatalog());
+        String result = importer.execute(null, sampleCoverage, SystemTestData.CITE_PREFIX, storeName,
+                "Buildings4", CRS.decode("EPSG:4326"), null, null);
+        // expect workspace:layername
+        assertEquals(result, SystemTestData.CITE_PREFIX + ":" + "Buildings4");
+    }
+
+    /**
+     * Test creating a vector store when a store name is specified but does not exist
+     */
+    @Test
+    public void testCreateDataStore() throws Exception {
+        FeatureTypeInfo ti = getCatalog().getFeatureTypeByName(getLayerId(SystemTestData.BUILDINGS));
+        SimpleFeatureCollection rawSource = (SimpleFeatureCollection) ti.getFeatureSource(null,
+                null).getFeatures();
+        ForceCoordinateSystemFeatureResults sampleData = new ForceCoordinateSystemFeatureResults(
+                rawSource, CRS.decode("EPSG:4326"));
+        String storeName = SystemTestData.CITE_PREFIX + "data";
+        DataStoreInfo storeInfo = catalog.getDataStoreByName(storeName);
+        assertNull("Store already exists " + storeInfo, storeInfo);
+        ImportProcess importer = new ImportProcess(getCatalog());
+        String result = importer.execute(sampleData, null, SystemTestData.CITE_PREFIX, storeName,
+                "Buildings5", CRS.decode("EPSG:4326"), null, null);
```

Comment:
Would be nice, but none of the other tests do that, so I'm ok with the new one ignoring the protocol too.